RecoHI/HiJetAlgos/plugins/GenericVoronoiAlgorithm.h

```diff
@@ -0,0 +1,143 @@
+#ifndef __GenericVoronoiAlgorithm_h__
+#define __GenericVoronoiAlgorithm_h__
+
+#include <cmath>
+
+#include <CGAL/Exact_predicates_inexact_constructions_kernel.h>
+#include <CGAL/Delaunay_triangulation_2.h>
+#include <CGAL/Voronoi_diagram_2.h>
+#include <CGAL/Delaunay_triangulation_adaptation_traits_2.h>
+#include <CGAL/Delaunay_triangulation_adaptation_policies_2.h>
+#include <CGAL/Polygon_2.h>
+
+#include <boost/multi_array.hpp>
+
+#include "DataFormats/Math/interface/LorentzVector.h"
+#include "DataFormats/Math/interface/deltaR.h"
+#include "RecoHI/HiJetAlgos/interface/UECalibration.h"
+
+class GenericVoronoiAlgorithm {
+protected:
+	//
+	typedef CGAL::Delaunay_triangulation_2<
+	CGAL::Exact_predicates_inexact_constructions_kernel>
+	delaunay_triangulation_t;
+	typedef delaunay_triangulation_t::Point point_2d_t;
+	class particle_t {
+	public:
+		math::PtEtaPhiELorentzVector momentum;
+		unsigned int reduced_particle_flow_id;
+		double area;
+		double momentum_perp_subtracted;
+		double momentum_perp_subtracted_unequalized;
+		std::set<std::vector<particle_t>::iterator> incident;
+		particle_t(math::PtEtaPhiELorentzVector p,
+				   unsigned int i, double a = NAN,
+				   double ps = NAN)
+			: momentum(p), reduced_particle_flow_id(i), area(a),
+			  momentum_perp_subtracted(ps),
+			  momentum_perp_subtracted_unequalized(ps),
+			  incident(std::set<std::vector<particle_t>::
+					   iterator>())
+		{
+		}
+		inline operator point_2d_t(void) const
+		{
+			return point_2d_t(momentum.Eta(), momentum.Phi());
+		}
+	};
+	typedef std::vector<particle_t> event_t;
+	// Remaining CGAL classes
+	typedef CGAL::Voronoi_diagram_2<
+		delaunay_triangulation_t,
+		CGAL::Delaunay_triangulation_adaptation_traits_2<
+			delaunay_triangulation_t>,
+		CGAL::
+		Delaunay_triangulation_caching_degeneracy_removal_policy_2<
+			delaunay_triangulation_t> > voronoi_diagram_t;
+	typedef CGAL::Polygon_2<
+		CGAL::Exact_predicates_inexact_constructions_kernel>
+	polygon_t;
+public:
+	static const size_t nreduced_particle_flow_id = 3;
+	static const size_t nfourier = 5;
+protected:
+	std::vector<double> _edge_pseudorapidity;
+	std::vector<double> _cms_hcal_edge_pseudorapidity;
+	std::vector<double> _cms_ecal_edge_pseudorapidity;
+	std::pair<double, double> _equalization_threshold;
+	bool _remove_nonpositive;
+	event_t _event;
+	//
+	double _radial_distance_square_max;
+	double _positive_bound_scale;
+	std::vector<bool> _active;
+	std::vector<std::pair<size_t, size_t> > _recombine;
+	std::vector<std::vector<size_t> > _recombine_index;
+	std::vector<std::vector<size_t> > _recombine_unsigned;
+	std::vector<double> _recombine_tie;
+	size_t _ncost;
+	std::vector<size_t> _nblock_subtract;
+	void *_lp_environment;
+	void *_lp_problem;
```

Comment:
This became a scoped variable and is inadvertently left in the class declaration. Fixed with the new commit.